### PR TITLE
[SC-148] Add Gnosis Chain domain

### PR DESCRIPTION
**Pull request by barrutko**

### CODE REVIEW

File: src/GnosisDomain.sol [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    function requiredSignatures() external view returns (uint256);
+}
+
+contract GnosisDomain is BridgedDomain {
+
+    bytes32 private constant USER_REQUEST_FOR_AFFIRMATION_TOPIC = keccak256("UserRequestForAffirmation(bytes32,bytes)");
+    bytes32 private constant USER_REQUEST_FOR_SIGNATURE_TOPIC   = keccak256("UserRequestForSignature(bytes32,bytes)");
+
+    IForeignAMB public immutable L1_AMB_CROSS_DOMAIN_MESSENGER;
+    IHomeAMB public immutable    L2_AMB_CROSS_DOMAIN_MESSENGER;
```
Comment (hexonaut) [inline src/GnosisDomain.sol:33]: These `immutable` needs to be removed to allow compiling.

**Review by hexonaut — CHANGES_REQUESTED** [review]
Looks good to merge once `immutable` is fixed.

File: src/GnosisDomain.sol (hunk 1 above)
Comment (barrutko) [reply src/GnosisDomain.sol]: Done

**Review by hexonaut — APPROVED** [review]

File: src/GnosisDomain.sol [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            address(validatorContract),
+            0x8a247e09a5673bd4d93a4e76d8fb9553523aa0d77f51f3d576e7421f5295b9bc,
+            0
+        );
+
+        vm.recordLogs();
+    }
+
+    function relayFromHost(bool switchToGuest) external override {
+        selectFork();
```
Comment (lucas-manuel) [inline src/GnosisDomain.sol:64]: Can we add comment to specify this is for gnosis?

File: src/GnosisDomain.sol [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        }
+    }
+
+    function removeFirst64Bytes(bytes memory inputData) public pure returns (bytes memory) {
+        bytes memory returnValue = new bytes(inputData.length - 64);
+        for (uint256 i = 0; i < inputData.length - 64; i++) {
+            returnValue[i] = inputData[i + 64];
+        }
+        return returnValue;
+    }
```
Comment (lucas-manuel) [inline src/GnosisDomain.sol:115]: Extra line here

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

**Review by lucas-manuel — APPROVED** [review]

**Review by hexonaut — APPROVED** [review]

**Pull request merged**